src/BuildScriptGeneratorCli/Commands/BuildScriptCommand.cs

```diff
@@ -4,31 +4,110 @@
+            var platformVersionOption = new Option<string>(OptionTemplates.PlatformVersion, OptionTemplates.PlatformVersionDescription);
+            var packageOption = new Option<bool>(OptionTemplates.Package, OptionTemplates.PackageDescription);
+            var osReqOption = new Option<string>(OptionTemplates.OsRequirements, OptionTemplates.OsRequirementsDescription);
+            var appTypeOption = new Option<string>(OptionTemplates.AppType, OptionTemplates.AppTypeDescription);
+            var buildCommandFileNameOption = new Option<string>(OptionTemplates.BuildCommandsFileName, OptionTemplates.BuildCommandsFileNameDescription);
+            var compressDestDirOption = new Option<bool>(OptionTemplates.CompressDestinationDir, OptionTemplates.CompressDestinationDirDescription);
+            var propertyOption = new Option<string[]>(OptionTemplates.Property, OptionTemplates.PropertyDescription);
+            var dynamicInstallRootDirOption = new Option<string>(OptionTemplates.DynamicInstallRootDir, OptionTemplates.DynamicInstallRootDirDescription);
+            var buildScriptOutputOption = new Option<string>(
+                name: "--output",
+                description: "The path that the build script will be written to. " +
+                          "If not specified, the result will be written to STDOUT.");
+
+            var command = new Command("build-script", "Generate build script to standard output.")
+            {
+                logOption,
+                debugOption,
+                sourceDirArgument,
+                platformOption,
+                platformVersionOption,
+                packageOption,
+                osReqOption,
+                appTypeOption,
+                buildCommandFileNameOption,
+                compressDestDirOption,
+                propertyOption,
+                dynamicInstallRootDirOption,
+                buildScriptOutputOption,
+            };
+
+            command.SetHandler(
+                (prop) =>
+                {
+                    var buildScriptCommand = new BuildScriptCommand(prop);
+                    return Task.FromResult(buildScriptCommand.OnExecute(console));
+                },
+                new BuildScriptCommandBinder(
+                    buildScriptOutputOption: buildScriptOutputOption,
+                    sourceDir: sourceDirArgument,
+                    platform: platformOption,
+                    platformVersion: platformVersionOption,
+                    package: packageOption,
+                    osRequirements: osReqOption,
+                    appType: appTypeOption,
+                    buildCommandFile: buildCommandFileNameOption,
+                    compressDestinationDir: compressDestDirOption,
+                    property: propertyOption,
+                    dynamicInstallRootDir: dynamicInstallRootDirOption,
+                    logPath: logOption,
+                    debugMode: debugOption));
```

Comment:
Similar comment as above: please consider aligning the names so that we're consistent everywhere. That way, it will reduce the uncertainty for devs working with this code late, as well as making searches in the code easier.

Reply:
I've just changed most of the alignment issue across different binders. For the alignments between parameter names and actual option names in single command file, I kept the naming convention so that it is obvious which type (option or argument) it is, which helps with the readability a bit of the code imo.

Reply:
OK, I had another comment along these lines that I added tonight. Please don't hesitate to let me know if it falls in the case you're mentioning here. Getting them aligned would have been great, but if it would reduce the readability overall I'm good with keeping as is.

Reply:
Yes, so i kept the type of the parameter into the variable name (e.g. option or argument). I think it'll be a reminder of what type of parameter is used to whoever is changing it.